### PR TITLE
Add release pipeline (updated from @hel0t)

### PR DESCRIPTION
**Pull request by ScrewTSW**
Example workflow run: https://github.com/ScrewTSW/Fika-ServerTools/actions/runs/10923433737
Example release: https://github.com/ScrewTSW/Fika-ServerTools/releases/tag/main

### CODE REVIEW

File: .github/workflows/tag-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      - name: Upload tar.gz artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: source_code.tar.gz
+          path: ./dist/source_code.tar.gz
+      - name: Upload zip artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: source_code.zip
+          path: ./dist/source_code.zip
```
Comment (ArchangelWTF) [inline .github/workflows/tag-and-release.yml]: Would this even be necessary? As far as I know Github automatically publishes the source code with a release.

File: .github/workflows/tag-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      - name: Download artifacts (tar.gz)
+        uses: actions/download-artifact@v4
+        with:
+          name: source_code.tar.gz
+          path: ./dist/
+      - name: Download artifacts (zip)
+        uses: actions/download-artifact@v4
+        with:
+          name: source_code.zip
+          path: ./dist/
```
Comment (ArchangelWTF) [inline .github/workflows/tag-and-release.yml]: Same as above

File: .github/workflows/tag-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+      - name: Upload source code (tar.gz)
+        uses: actions/upload-release-asset@v1.0.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./dist/source_code.tar.gz
+          asset_name: source_code.tar.gz
+          asset_content_type: application/gzip
```
Comment (ArchangelWTF) [inline .github/workflows/tag-and-release.yml]: Same as above

**Review by ArchangelWTF — COMMENTED** [review]
Thanks for the PR! I've got a few comments about some of the things in this pipeline

**Review by ArchangelWTF — APPROVED** [review]

File: .github/workflows/tag-and-release.yml (hunk 1 above)
Comment (ScrewTSW) [reply .github/workflows/tag-and-release.yml]: you were correct, fixed

File: .github/workflows/tag-and-release.yml (hunk 2 above)
Comment (ScrewTSW) [reply .github/workflows/tag-and-release.yml]: fixed

File: .github/workflows/tag-and-release.yml (hunk 3 above)
Comment (ScrewTSW) [reply .github/workflows/tag-and-release.yml]: fixed

**Comment by Greelan** [comment]
The server on Linux still seems to be looking for the windows exe?

`[FikaServerTools] File not found: /AMP/tarkov/server/user/mods/fika-server/FikaServerTools.exe`

**Comment by ScrewTSW** [comment]
> The server on Linux still seems to be looking for the windows exe?
> 
> `[FikaServerTools] File not found: /AMP/tarkov/server/user/mods/fika-server/FikaServerTools.exe`

Yes, that's correct unfortunately, but you can start the FikaServerTools manually alongside the SPT server and ignore the error message.

It actually works when set up correctly, tested it on my machine.

But I am curious, how did you manage to get a Linux build of the SPT server? And do the mods work with it?

**Comment by trippyone** [comment]
> The server on Linux still seems to be looking for the windows exe?
> 
> `[FikaServerTools] File not found: /AMP/tarkov/server/user/mods/fika-server/FikaServerTools.exe`

Yes. This PR allows to build a Linux executable of Fika-ServerTools. But the fika-server plugin is the one launching the executable and support for Linux was not added to it.

**Comment by Greelan** [comment]
Got it. I wasn't looking for a standalone build but appreciate the update.

**Comment by ArchangelWTF** [comment]
> > The server on Linux still seems to be looking for the windows exe?
> > `[FikaServerTools] File not found: /AMP/tarkov/server/user/mods/fika-server/FikaServerTools.exe`
> 
> Yes. This PR allows to build a Linux executable of Fika-ServerTools. But the fika-server plugin is the one launching the executable and support for Linux was not added to it.

Support is added for it in 3.10

**Comment by Greelan** [comment]
Running SPT 3.10.0 and Fika 2.3.0 on Linux and I still have issues loading FikaServerTools:

`Error: spawn /AMP/tarkov/server/user/mods/fika-server/FikaServerTools EACCES
Error: spawn /AMP/tarkov/server/user/mods/fika-server/FikaServerTools EACCES
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
Error: spawn /AMP/tarkov/server/user/mods/fika-server/FikaServerTools EACCES
Error: spawn /AMP/tarkov/server/user/mods/fika-server/FikaServerTools EACCES
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections`

Would welcome any thoughts.

**Comment by Greelan** [comment]
Nvm, just need to `chmod +x`

**Pull request merged**